doc/contribute/guidelines.rst

```diff
@@ -536,12 +536,21 @@ reference manuals, etc.
 Coding Style
 ============
 
+.. note::
+   Coding style is enforced on any new or modified code, but contributors are
+   not expected to correct the style on existing code that they are not
+   modifying.
+
 .. _Linux kernel coding style:
    https://kernel.org/doc/html/latest/process/coding-style.html
 
+.. _snake case:
+   https://en.wikipedia.org/wiki/Snake_case
+
 In general, follow the `Linux kernel coding style`_, with the following
-exceptions:
+exceptions and clarifications:
 
+* Use `snake case`_ for code and screaming snake case for macros and constants.
```

Comment:
Please remove this line, a quick `grep -r '#define [a-z]*(' *` on the codebase shows that cannot be a general rule.

Reply:
What do you mean? This is only for C code, do you mean that we have CamelCase or others in the codebase? I am all for consensus, but Zephyr is snake case, I don't think anyone will dispute that?

Reply:
Oh I see what you mean now, about the macros. My mistake @Laczen. Let me try and simplify this by just removing the statement about macros.

Reply:
@Laczen thanks for the comment, I  have now simplified it.